client/blocks/login/index.jsx

```diff
@@ -197,6 +201,24 @@ class Login extends Component {
 		} );
 	};
 
+	getSignupUrl = () => {
+		if ( this.props.signupUrl ) {
+			return this.props.signupUrl;
+		}
+
+		const { currentQuery, currentRoute, oauth2Client, locale, isGutenboarding } = this.props;
+		const { pathname } = getUrlParts( window.location.href );
+
+		return getSignupUrl(
+			currentQuery,
+			currentRoute,
+			oauth2Client,
+			locale,
+			pathname,
+			isGutenboarding
+		);
+	};
```

Comment:
Is this the best way to produce a signup URL?

Reply:
I think it's a little confusing that this `<Login>` block both accepts a `signupUrl` prop, and also has the ability to generate its own signup URL using this `getSignupUrl` method. Should this method be used everywhere the `signupUrl` prop is used? so it's more like providing a default value for the prop?

Another option might be to remove all this `getSignupUrl` logic from the `<Login>` component. Since all it ever does with this logic is just pass it on to the `<ErrorNotice>`, maybe the logic should just be moved inside `<ErrorNotice>` instead?

Reply:
> Yep this is the helper to use. There's quite a bit that goes into calculating the signup URL, and the user could have landed on the login page from lots of different places, so it's good to maintain all that context when we send them to the signup page

Great, thanks for clarifying.

> I think it's a little confusing that this `<Login>` block both accepts a `signupUrl` prop, and also has the ability to generate its own signup URL using this `getSignupUrl` method. Should this method be used everywhere the `signupUrl` prop is used? so it's more like providing a default value for the prop?

I think it's odd too, and I'm not sure what the original intent was. However, I feel reasonably confident @zaguiini had a rationale for it.

I'm open to your opinion on what should be done instead.

Reply:
@p-jackson Also, when testing in my local environment, `getSignupUrl` seems to produce `/start` instead of `/start/user`. Does this make sense to you? Anything we need to do to resolve?

Reply:
> I'm open to your opinion on what should be done instead.

I was thinking something like this https://github.com/Automattic/wp-calypso/pull/66148/files that moves the logic for generating the signup URL into the `<ErrorNotice>` since it seemed to be the only thing that used it.
But now that I've tried it I don't like it. It requires passing the `locale` and `isGutenboarding` props into the `<ErrorNotice>` component anyway, so I think it's just as tidy without my changes.

> `getSignupUrl` seems to produce `/start` instead of `/start/user`

Yep that's good. `/start` is the usual link we would use to start off the flow. The signup flow will decide whether they need to create an account or not and redirect them to the `user` step if necessary.